### DISCUSSION TITLE
在umijs4的base配置项中设置了部署根目录，但是在build输出中无效。

### DISCUSSION THREAD

**Discussion by jerrylee2013** [umijs/umi#10751] [Q&A] [answered]
比如我在base中设置根目录是/test（base: '/test'），把max build出来的内容拷贝到服务器下的test目录后，直接访问/test，会提示umi.js找不到。
在输出产物中index.html看到<script src="/umi.js"></script>，手动修改成<script src="/test/umi.js"></script>后才能加载。但是相关的依赖产物却404了。请问这个是不是问题，还是我对base的配置用法不对？

我试过(base:'test'),(base:'/test/')，结果都一样。

**Comment by jerrylee2013** [comment]
补充一下，我在"@umijs/max": "^4.0.57"和^4.0.56结果都一样。 dev环境是ok的，但是max build出来的产物就不行

**Comment by fz6m** [comment, marked as answer]
`base` 是部署的时候 nginx 反代的那个路径，比如 `/base` 的请求全打到 `index.html` ，这个时候 `/base/` 就是根路由。

`publicPath` 是静态资源提供的路径，他不一定和你的 `base` 相同，要看你部署方式，从哪个 prefix 能访问到资源，就配到 `publicPath` 上。

这俩搞清楚了就能成功在子目录部署了。

**Reply by jerrylee2013** [reply]
哦，我搞错了，谢谢澄清，确实用publicPath就ok了
